### DISCUSSION TITLE
Unable to ping across site-to-site VPN to Check Point

### DISCUSSION THREAD

**Discussion by Eric-MDI** [strongswan/strongswan#485] [Q&A]
Hello,
  I am very very new to all of this so please be easy on me and when providing instructions if you can provide the full command I am to run that would be very helpful. 

  We are trying to setup a site-to-site VPN between our gateway and a check point gateway. Our side is natted due to a conflict with our native subnet on their end. I can get the VPN tunnel to come up. Or, at least appears to come up but we cannot ping devices on their side and get a responce. They claim they are seeing our icmp packets come in and are responding to them but we never receive the replies. When I perform a tcpdump looking for all icmp traffic I do not see anything from them. I am guessing we are missing something on our end related to nat that is not properly receiving the traffic they are sending and delivering it back to the originating source.

  Any help would be greatly appreciated.

Local peer ip - 75.18.217.170
Local subnet - 192.168.1.0/24
Nat subnet - 172.201.47.0/24

Remote peer ip - 52.11.10.92
Remote subnets - 10.122.106.0/27, 10.122.106.32/27, 10.122.50.0/27, 10.122.50.32/27
IKE ID - 10.122.0.24

  Attached are the files requested.

Thank you and have a great day.

[ip address.txt](https://github.com/strongswan/strongswan/files/6735348/ip.address.txt)
[ip route show table all.txt](https://github.com/strongswan/strongswan/files/6735349/ip.route.show.table.all.txt)
[ip6tables-save.txt](https://github.com/strongswan/strongswan/files/6735350/ip6tables-save.txt)
[ipsec.conf.txt](https://github.com/strongswan/strongswan/files/6735351/ipsec.conf.txt)
[iptables-save.txt](https://github.com/strongswan/strongswan/files/6735352/iptables-save.txt)
[spsec statusall.txt](https://github.com/strongswan/strongswan/files/6735353/spsec.statusall.txt)
[strongswan.conf.txt](https://github.com/strongswan/strongswan/files/6735354/strongswan.conf.txt)

**Comment by tobiasbrunner** [comment]
Maybe try adding a `NETMAP` rule in `PREROUTING` to map `172.201.47.0/24` back to `192.168.1.0/24` (in particular if they should be able to initiate traffic to hosts on your end).

**Reply by tobiasbrunner** [reply]
You already have a rule in `POSTROUTING`, do it basically the same way just reversed (not sure if decrypted packets have the VTI device associated, so if this doesn't work, try without `-i`):

    iptables -t nat -A PREROUTING -d 172.201.47.0/24 -i vti65 -j NETMAP --to 192.168.1.0/24

**Reply by tobiasbrunner** [reply]
No restarts should be necessary. It's difficult to say what exactly the problem is. You should check traffic counters (`ipsec statusall`) to see if a response is received and decrypted (also check the counters in `/proc/net/xfrm_stat` if available). Using the TRACE target to debug the packet flow might also help (maybe [this](https://github.com/commonism/iptables-trace) could be useful).

> Also, does it matter that the Check Point side of the tunnel claims to have setup their side up as policy based?

That's generally a local decision. But configs (e.g. traffic selectors) could be quite different for the two approaches. And the use of VTIs on your end could lead to issues in itself in particular if combined with NATs etc. (see [here](https://wiki.strongswan.org/projects/strongswan/wiki/RouteBasedVPN) for general information about route-based VPNs).

**Reply by tobiasbrunner** [reply]
> My concern with doing away with the VTI interface is making sure only the traffic that is supposed to go across the tunnel does and all other traffic goes out normally. I think this is done with split tunneling but when looking into how to implement I am unable to follow exactly what needs to be done and what specific commands I need to enter to get things setup properly. Any insight into that?

You'd do that by negotiating separate CHILD_SAs with traffic selectors (`left|rightsubnet`) that specify exactly the traffic you want to tunnel (with IKEv1 you can't have multiple traffic selectors per CHILD_SA, or narrowing, which would make this simpler, so the subnets have to match exactly on both ends). The other peer actually does negotiate such CHILD_SAs as you can see in `ipsec statusall`. So instead of

    leftsubnet=0.0.0.0/0
    rightsubnet=0.0.0.0/0

You'd specify the actual subnets, for instance,

    leftsubnet=172.201.47.0/24
    rightsubnet=10.122.106.0/27

and then you'd add additional conn sections (you can use `also=<nameofbaseconn>` in those) for each of the other `rightsubnet` values.

**Reply by tobiasbrunner** [reply]
>  When I look at the counters they seem to go up when I send out a packet but it does not come back.

Then there might be an issue on their end (e.g. with routing to your virtual subnet `172.201.47.0/24`).

**Reply by tobiasbrunner** [reply]
> If I have a continuous ping going to 10.122.50.60 and 10.122.106.15 from a computer on my side of the tunnel and then restart the gateway the ping will succeed in getting replies until I restart that computer. No other computer will be able to get replies during this.

Not sure if I understand you correctly. Are you saying the VPN connection is actually working correctly if pinging from that particular host? And what's the relevance of the gateway restart? Does it work before, after, during that restart? And what do you mean when you say no other host gets replies "during this"? Do they get them at other times? And during what exactly? Also, what IP addresses do all of these hosts have?

**Reply by tobiasbrunner** [reply]
Can you post the output of `ipsec statusall` after the restart (when that one client can ping)? There might be a very specific SA established for only that client.

**Comment by Eric-MDI** [comment]
Hello,
  Thank you for the suggestion. I am not sure how to do that. Do you happen to know the exact command I should enter?

Thank you.
On Jun 30, 2021, at 3:47 AM, Tobias Brunner ***@***.******@***.***>> wrote:
-EXTERNAL-


Maybe try adding a NETMAP rule in PREROUTING to map 172.201.47.0/24 back to 192.168.1.0/24 (in particular if they should be able to initiate traffic to hosts on your end).

**Comment by Eric-MDI** [comment]
Hello,
  Thank you very much for the reply. I tried it with and without the -i and neither allowed me to ping successfully. Do I need to restart the gateway or run an ipsec restart command to get it to pass traffic through or should it just start working? Also, does it matter that the Check Point side of the tunnel claims to have setup their side up as policy based?

Thank you.

**Comment by Eric-MDI** [comment]
Thank you for your help. Really appreciate it. I'll take a look at the links you provided. My concern with doing away with the VTI interface is making sure only the traffic that is supposed to go across the tunnel does and all other traffic goes out normally. I think this is done with split tunneling but when looking into how to implement I am unable to follow exactly what needs to be done and what specific commands I need to enter to get things setup properly. Any insight into that?

Thank you.
On Jul 1, 2021, at 2:07 AM, Tobias Brunner ***@***.******@***.***>> wrote:
-EXTERNAL-


No restarts should be necessary. It's difficult to say what exactly the problem is. You should check traffic counters (ipsec statusall) to see if a response is received and decrypted (also check the counters in /proc/net/xfrm_stat if available). Using the TRACE target to debug the packet flow might also help (maybe this<https://github.com/commonism/iptables-trace> could be useful).

Also, does it matter that the Check Point side of the tunnel claims to have setup their side up as policy based?

That's generally a local decision. But configs (e.g. traffic selectors) could be quite different for the two approaches. And the use of VTIs on your end could lead to issues in itself in particular if combined with NATs etc. (see here<https://wiki.strongswan.org/projects/strongswan/wiki/RouteBasedVPN> for general information about route-based VPNs).

**Comment by Eric-MDI** [comment]
Hello,
  Thank you for your reply. I gave it a try and still no traffic comes back. When I look at the counters they seem to go up when I send out a packet but it does not come back. I am not seeing any imcp packets coming back from their end. I tried to setup the VPN without the VTI interface but that didn’t work either.

Thank you.

**Comment by Eric-MDI** [comment]
Thank you. Here is a strange thing that I just tested. If I have a continuous ping going to 10.122.50.60 and 10.122.106.15 from a computer on my side of the tunnel and then restart the gateway the ping will succeed in getting replies until I restart that computer. No other computer will be able to get replies during this. Seem odd that I can get a reply on that one computer. I have been trying to get them to check their side again for weeks but they keep saying it has to be something on my end.

**Comment by Eric-MDI** [comment]
Sorry to be confusing. I will try to explain it better.

Host 1 has an ip address of 192.168.1.123. I can have a continuous ping going to 10.122.50.60 and 10.122.106.15 from this host. No replies are received. If I restart the gateway with the continuous ping going once the gateway comes back up host 1 starts to receive replies successfully.

Host 2 has an ip address of 192.168.1.125. After gateway restart I begin a continuous ping to 10.122.50.60 and 10.122.106.15. No replies are received even though host 1 is still receiving replies successfully.

If I now restart host 1 and try to ping 10.122.50.60 and 10.122.106.15 no replies will be received again.

Very strange behavior. Oh, and this same thing is happening at two locations so it is not isolated to just this location. The equipment at both location is the same and the VPN is to the same third party company. The hope is that we can figure out how to fix location 1 so that I can also implement the solution at location 2.

**Comment by Eric-MDI** [comment]
Hello,
  Will do. Right now the office is in use so I can't reboot the gateway. I will do so after hours and post the results.

Thank you for all of your help.
On Jul 2, 2021, at 7:37 AM, Tobias Brunner ***@***.******@***.***>> wrote:
-EXTERNAL-


Can you post the output of ipsec statusall after the restart (when that one client can ping)? There might be a very specific SA established for only that client.

**Comment by Eric-MDI** [comment]
Hello,
  Sorry for the slow response. This office has been closed due to a gas leak since 7/2 in the afternoon. Once they are allowed back in the building and power is turned back on I will be able to run the ipsec statusall and will report what it says.

Thank you.
